File: unity/BlackjackUnity/Assets/Screentip.cs
/*
  Blackjack
  Copyright 2015 Sam Saint-Pettersen
  Released under the MIT/X11 License.

  C#/XNA implementation
*/
using UnityEngine;
using System;
using System.Collections;

/// <summary>
/// Screentip implements a title and message box.
/// </summary>
public class Screentip {
    private bool debug;
    private int posX;
    private int posY;
    private string title;
    private string msg;

    /// <summary>
    /// Constructor for screentip.
    /// </summary>
    /// <param name="debug">Enable debug messages?</param>
    /// <param name="posX">X position for screentip.</param>
    /// <param name="posY">Y position for screentip.</param>
    public Screentip(bool debug, int posX, int posY) {
        this.debug = debug;
        this.posX = posX;
        this.posY = posY;
        Debugger.Emit(this.debug, String.Format("Created screentip at {0},{1}", posX, posY));
    }

    /// <summary>
    /// Emit a title and message.
    /// </summary>
    /// <param name="title">Title to emit.</param>
    /// <param name="message">Message to emit.</param>
    public void Emit(string title, string message) {
        this.Clear();
        this.title = title;
        this.msg = message;
    }

    /// <summary>
    /// Clear the screentip.
    /// </summary>
    public void Clear() {
        this.title = "";
        this.msg = "";
    }

    /// <summary>
    /// Draw the screentip.
    /// </summary>
    public void Draw() {
        Rect rect1 = new Rect(this.posX, this.posY, 300, 20);
        Rect rect2 = new Rect(this.posX - 45, this.posY + 20, 300, 20);
        GUI.Label(rect1, this.title);
        GUI.Label(rect2, this.msg);
    }
}

File: html5/enchant.js_ts/screentip.ts
/*
	Blackjack
	Copyright 2015 Sam Saint-Pettersen
	Released under the MIT/X11 License.
	
	HTML5/enchant.js TypeScript implementation
*/

/// <reference path="enchant.d.ts" />
/// <reference path="debug.ts" />

class Screentip {
	private title: Label;
	private msg: Label;

	/**
	 * Screentip implements a ttile and message box.
	 * @constructor
	 * @param {boolean} debug - Enable debug messages?
	 * @param {number} posX - X position for screentip.
	 * @param {number} posY - Y position for screentip.
	*/
	constructor(debug: boolean, posX: number, posY: number) {
		this.title = new Label('');
		this.msg = new Label('');
		this.title.color = 'rgb(255, 255, 255)';
		this.title.font = '10pt verdana, sans-serif';
		this.title.x = posX;
		this.title.y = posY;
		this.msg.color = this.title.color;
		this.msg.font = this.title.font;
		this.msg.x = posX - 45;
		this.msg.y = posY - 20;

		Debug.emit(debug, "Created screentip at ${posX},${posY}");
	}

	/**
	 * Emit a title and message.
	 * @param {string} title - Title to emit.
	 * @param {string} message - Message to emit.
	*/
	public emit(title: string, message: string): void {
		this.clear();
		this.title.text = title;
		this.msg.text = message;
	}

	/**
	 * Clear the screentip.
	*/
	public clear(): void {
		this.title.text = '';
		this.msg.text = '';
	}

	/**
	 * Draw the screentip.
	 * @returns {Label[]} The screentip.
	*/
	public draw(): Label[] {
		if (this.msg.text == null) this.msg.text = '';
		if (this.title.text == null) this.title.text = '';
		return [this.title, this.msg];
	}
}
